File: Core/Entities/ReviewCommentFeedback.cs
namespace Core.Entities
{
    public class ReviewCommentFeedback : BaseEntity
    {
        public Guid? ParentId { get; set; }
        public ReviewCommentFeedback? Parent { get; set; }
        public List<ReviewCommentFeedback> Replies { get; set; } = [];
        public string Comment { get; set; } = null!;
        public Guid ReviewPostId { get; set; }
        public ReviewPost ReviewPost { get; set; } = null!;
        public List<ReviewReaction> Reactions { get; set; } = [];
    }
}

File: client/src/app/shared/models/review/review-comment-feedback.ts
import { BaseModel } from "../base";
import { ReviewPostModel } from "./review-post";
import { ReviewReactionModel } from "./review-reaction";

export interface ReviewCommentFeedbackModel extends BaseModel {
    parentId: string | null;
    parent: ReviewCommentFeedbackModel | null;
    replies: ReviewCommentFeedbackModel[];
    comment: string;
    reviewPostId: string;
    reviewPost: ReviewPostModel;
    reactions: ReviewReactionModel[];
}